pkg/asset/installconfig/ovirt/validation.go

```diff
@@ -63,6 +66,129 @@ func validateMachinePool(con *ovirtsdk.Connection, child *field.Path, pool *ovir
+					fmt.Sprintf("Invalid affinity group %v: name must be not empty", ag.Name)))
+		}
+		if ag.Priority < 0 || ag.Priority > 5 {
+			allErrs = append(
+				allErrs,
+				field.Invalid(fldPath, ag,
+					fmt.Sprintf(
+						"Invalid affinity group %v: priority value must be between 0-5 found priority %v",
+						ag.Name,
+						ag.Priority)))
+		}
+	}
+	return allErrs
+}
+
+// validateExistingAffinityGroup checks that there is no affinity group with the same name in the cluster
+func validateExistingAffinityGroup(con *ovirtsdk.Connection, platform ovirt.Platform, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	res, err := con.SystemService().ClustersService().
+		ClusterService(platform.ClusterID).AffinityGroupsService().List().Send()
+	if err != nil {
+		allErrs = append(
+			allErrs,
+			field.InternalError(
+				fldPath,
+				errors.Errorf("failed listing affinity groups for cluster %v", platform.ClusterID)))
+	}
+	for _, ag := range res.MustGroups().Slice() {
+		for _, agNew := range platform.AffinityGroups {
+			if ag.MustName() == agNew.Name {
+				allErrs = append(
+					allErrs,
+					field.Invalid(
+						fldPath,
+						ag,
+						fmt.Sprintf("affinity group %v already exist in cluster %v", agNew.Name, platform.ClusterID)))
+			}
+		}
+	}
+	return allErrs
+}
+
+// validateAffinityGroupDuplicate checks that there is no duplicated affinity group with different fields
+func validateAffinityGroupDuplicate(agList []*ovirt.AffinityGroup) field.ErrorList {
+	allErrs := field.ErrorList{}
+	for i, ag1 := range agList {
+		for _, ag2 := range agList[i+1:] {
+			if ag1.Name == ag2.Name {
+				if ag1.Priority != ag2.Priority ||
+					ag1.Description != ag2.Description ||
+					ag1.Enforcing != ag2.Enforcing {
+					allErrs = append(
+						allErrs,
+						&field.Error{
+							Type: field.ErrorTypeDuplicate,
+							BadValue: errors.Errorf("Error validating affinity groups: found same "+
+								"affinity group defined twice with different fields %v anf %v", ag1, ag2)})
+				}
+			}
+		}
+	}
+	return allErrs
+}
+
+func validateClusterResources(con *ovirtsdk.Connection, ic *types.InstallConfig, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	mAgReplicas := make(map[string]int)
+	for _, agn := range ic.ControlPlane.Platform.Ovirt.AffinityGroupsNames {
+		mAgReplicas[agn] = mAgReplicas[agn] + int(*ic.ControlPlane.Replicas)
+	}
+	for _, compute := range ic.Compute {
+		for _, agn := range compute.Platform.Ovirt.AffinityGroupsNames {
+			mAgReplicas[agn] = mAgReplicas[agn] + int(*compute.Replicas)
+		}
+	}
+
+	clusterName, err := GetClusterName(con, ic.Ovirt.ClusterID)
+	if err != nil {
+		allErrs = append(allErrs, field.InternalError(fldPath, err))
+	}
+	hosts, err := FindHostsInCluster(con, clusterName)
+	if err != nil {
+		allErrs = append(allErrs, field.InternalError(fldPath, err))
+	}
+	for _, ag := range ic.Ovirt.AffinityGroups {
+		if _, found := mAgReplicas[ag.Name]; found {
```

Comment:
Is there validation somewhere to ensure that the affinity group selected for a machine pool either is included in the affinity groups created by the installer or already exists in the oVirt cluster?

Reply:
Added one